app/Controller/BankTransactionsController.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ *
+ * PHP 5
+ *
+ * Copyright (C) HMS Team
+ *
+ * Licensed under The MIT License
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     HMS Team
+ * @package       app.Controller
+ * @license       MIT License (http://www.opensource.org/licenses/mit-license.php)
+ */
+
+App::uses('AppController', 'Controller');
+App::uses('CakeEmail', 'Network/Email');
+    
+/**
+ * Controller for the Bank Transactions, handles upload of CSV's and 
+ * showing an Accounts Transaction history
+ */
+class BankTransactionsController extends AppController {
+
+/**
+ * List of models this controller uses.
+ * @var array
+ */
+    public $uses = array('BankTransaction', 'Member', 'Meta');
+    
+/** 
+ * Test to see if a user is authorised to make a request.
+ *
+ * @param array $user Member record for the user.
+ * @param CakeRequest $request The request the user is attempting to make.
+ * @return bool True if the user is authorised to make the request, otherwise false.
+ * @link http://api20.cakephp.org/class/cake-request
+ */
+	public function isAuthorized($user, $request) {
+		if (parent::isAuthorized($user, $request)) {
+			return true;
+		}
+
+        $memberId = $this->Member->getIdForMember($user);
+        
+		$authGranted = false;
+
+		// Only history page implemented so far
+		if ($request->params['action'] == 'history') {
+            // Get the member_id details have been requested for & the logged in users member_id
+            if (isset($request->params['pass'][0])) {
+                $reqMemberId = $request->params['pass'][0];
+            } else {
+                $reqMemberId = $memberId;
+            }
+
+            // Allow everyone to view their own transaction history
+            if ($reqMemberId == $memberId) {
+                $authGranted = true;
+            } elseif ($this->Member->GroupsMember->isMemberInGroup( $memberId, Group::MEMBERSHIP_ADMIN )) {
+                // Only allow 'Full Access' (via parent::isAuthorized) and 'Membership Admins' to view the transaction history of others
+                $authGranted = true;
```

Comment:
Need to review this, not sure I want membership team admins seeing how much everyone pays, should be reserved to trustees only